src/activations.jl

```diff
@@ -734,6 +741,18 @@ end
 
 sigmoid_fast(x::Float16) = sigmoid(x)  # sigmoid_fast is extremely badly behaved at large x
 
+"""
+    NNlib.fast_act(f, [x::AbstractArray])
+
+Replaces `f == tanh` with [`tanh_fast`](@ref), etc.
+
+Takes an optional 2nd argument, so that you can disable
+this replacement for some array or element types.
+"""
+@inline fast_act(f::F, ::AbstractArray = 1:0) where {F<:Function} = f
```

Comment:
:eyes:  I've never seen this `::AbstractArray = 1:0` pattern before, what does it do?

Reply:
My thinking is that we may later decide that it's better to skip `tanh_fast` on the GPU. I can't measure a difference so who knows. To do that, we can add a method in NNlibCUDA like `fast_act(::typeof(tanh), ::CuArray) = tanh`. Provided you call this like `fast = NNlib.fast_act(fun, x)` with an example of the array you plan to broadcast it over.

Reply:
My branch for this is  https://github.com/mcabbott/Flux.jl/commit/217eb1ab55087c034524162ab901d430940162eb
which I see I did on top of https://github.com/FluxML/Flux.jl/pull/1761